Author: Reinhard Tartler
Description: Use pkg-config for detecting tinyxml
Forwarded-Upstream: no
Bug-Debian: #713515

The detection of tinyxml fails in debian because the library is no
longer installed into /usr/lib, is now in /usr/lib/$(ARCH_TRIPLET), for
instance /usr/lib/x86_64-linux-gnu. The upstream included
Find_tinyxml.cmake detects the Headers correctly, but breaks the build
for failing to locate the libraries.

Tinyxml ships a working pkg-config file, so let's use that one for now.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,7 +48,7 @@ INCLUDE(cmake_find_modules/Find_dl.cmake
 INCLUDE(cmake_find_modules/pch_support.cmake)
 INCLUDE(CMakeDetermineSystem)
 INCLUDE(TestBigEndian)
-
+INCLUDE(FindPkgConfig)
 
 #setup the endianess if not already defined in toolchain file
 if(NOT DEFINED NO_ENDIANESS)
@@ -101,10 +101,7 @@ ELSE($ENV{COMPILE_STATIC} MATCHES "1")
 		ADD_SUBDIRECTORY(lua lua)
 	ENDIF(NOT LUA_FOUND)
 
-	INCLUDE(cmake_find_modules/Find_tinyxml.cmake)
-	IF(NOT TINYXML_FOUND)
-		ADD_SUBDIRECTORY(tinyxml tinyxml)
-	ENDIF(NOT TINYXML_FOUND)
+       PKG_CHECK_MODULES(TINYXML REQUIRED tinyxml)
 ENDIF($ENV{COMPILE_STATIC} MATCHES "1")
 
 #fine tunning
--- a/common/CMakeLists.txt
+++ b/common/CMakeLists.txt
@@ -6,7 +6,7 @@ FILE(GLOB_RECURSE SOURCES "${CRTMPSERVER
 
 #prepare the include directories
 INCLUDE_DIRECTORIES(
-	${TINYXML_INCLUDE_PATH}
+	${TINYXML_INCLUDE_DIRS}
 	${OPENSSL_INCLUDE_PATH}
 	${LUA_INCLUDE_PATH}
 	${CRTMPSERVER_SOURCES_ROOT}/common/include
@@ -19,7 +19,7 @@ IF(${LIB_TYPE} MATCHES "SHARED")
 	TARGET_LINK_LIBRARIES(
 		${PROJECT_NAME} 
 		${LUA_LIBRARY_PATH} 
-		${TINYXML_LIBRARY_PATH} 
+		${TINYXML_LIBRARIES}
 		${OPENSSL_LIBRARY_PATH} 
 		${CRYPTO_LIBRARY_PATH} 
 		${Z_LIBRARY_PATH}
--- a/crtmpserver/CMakeLists.txt
+++ b/crtmpserver/CMakeLists.txt
@@ -28,7 +28,7 @@ TARGET_LINK_LIBRARIES(${PROJECT_NAME}
 	${APPS_LIBS}
 	thelib
 	common
-	${TINYXML_LIBRARY_PATH}
+	${TINYXML_LIBRARIES}
 	${LUA_LIBRARY_PATH}
 	${OPENSSL_LIBRARY_PATH} 
 	${CRYPTO_LIBRARY_PATH}
